aiosmtpd/tests/test_smtp.py

```diff
@@ -941,6 +970,26 @@ def test_auth_good_credentials(self):
             )
             assert_auth_success(self, code, response)
 
+    def test_authplain_goodcreds_sanitized_log(self):
+        oldlevel = mail_logger.getEffectiveLevel()
+        mail_logger.setLevel(logging.DEBUG)
+        with SMTP(*self.address) as client:
+            client.ehlo('example.com')
+            with self.assertLogs(mail_logger, level="DEBUG") as cm:
+                code, response = client.docmd(
+                    'AUTH PLAIN ' +
+                    b64encode(b'\0goodlogin\0goodpasswd').decode()
+                )
+        mail_logger.setLevel(oldlevel)
+        interestings = [
+            msg for msg in cm.output if "AUTH PLAIN" in msg
+        ]
+        assert len(interestings) == 2
+        assert interestings[0].startswith("DEBUG:")
+        assert interestings[0].endswith("b'AUTH PLAIN ********\\r\\n'")
+        assert interestings[1].startswith("INFO:")
+        assert interestings[1].endswith("b'AUTH PLAIN ********'")
```

Comment:
Ensure that log is sanitized.

So in the future, if a maintainer somehow forgets that those particular lines in `smtp.SMTP` needs sanitizing, the test case will fail.

Reply:
**praise**: I *love* providing tests for future maintainers. Usually because I'm that future maintainer :trollface: